app/server.go

```diff
@@ -502,8 +502,14 @@ func NewServer(options ...Option) (*Server, error) {
 	if appErr != nil {
 		mlog.Error("Problem with file storage settings", mlog.Err(appErr))
 	} else {
-		if nErr := backend.TestConnection(); nErr != nil {
-			mlog.Error("Problem with file storage settings", mlog.Err(nErr))
+		nErr := backend.TestConnection()
+		if nErr != nil {
+			if errors.Is(nErr, filestore.ErrNoS3Bucket) {
+				nErr = backend.(*filestore.S3FileBackend).MakeBucket()
```

Comment:
I'm just curious but what is this notation called `.(*filestore.S3FileBackend).` . Interested to understand how you can call the function `MakeBucket` when it isn't defined in the `FileBackend` interface. 

Also, let's add a test for this where no S3 bucket is found and it attempts to make a new one. Ensure that this really does happen.

Reply:
It's a [type assertion](https://tour.golang.org/methods/15).

I'll make sure to add a test for the server startup.